sdk/identity/identity/src/msal/nodeFlows/nodeCommon.ts

```diff
@@ -72,15 +74,16 @@ export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
   protected confidentialApp: msalNode.ConfidentialClientApplication | undefined;
   protected msalConfig: msalNode.Configuration;
   protected clientId: string;
+  protected tenantId: string;
   protected identityClient?: IdentityClient;
   protected requiresConfidential: boolean = false;
   protected azureRegion?: string;
-
   protected createCachePlugin: (() => Promise<msalCommon.ICachePlugin>) | undefined;
 
   constructor(options: MsalNodeOptions) {
     super(options);
     this.msalConfig = this.defaultNodeMsalConfig(options);
+    this.tenantId = resolveTenantId(options.logger, options.tenantId, options.clientId);
```

Comment:
Does `resolveTenantId` take the new `tenantId` value on `GetTokenOptions` into account along with the `tenantId` set by the credential?

Reply:
resolveTenantId has nothing to do with GetTokenOptions, you can assume it simply passes the credential constructor tenant through. For getToken we have the validateMultiTenantRequest call below in this file.